File: backend/src/main/java/indiana/indi/indiana/dto/games/GameFullDto.java
package indiana.indi.indiana.dto.games;

import indiana.indi.indiana.dto.categories.CategoryForGameDto;

import java.math.BigDecimal;
import java.util.List;

public record GameFullDto(
        Long id,
        String title,
        String details,
        String imageUrl,
        String gameFileUrl,
        Long authorId,
        List<CategoryForGameDto> categories,
        BigDecimal price) {
}

File: frontend/src/types/GameFullDto.ts
export type GameFullDto = {
    id: number;
    title: string;
    details: string;
    imageUrl: string;
    gameFileUrl: string;
    authorId: number;
    categories: Category[];
    price: number;
    isFavorite: boolean;
    isInCart: boolean;
    isPurchased: boolean;
}
